tests/forms_tests/tests/test_forms.py

```diff
@@ -4795,17 +4796,96 @@ def __init__(self, *args, **kwargs):
+                colour = CharField(max_length=10)
+
+            f = TestForm()
+            self.assertHTMLEqual(
+                f.as_div(),
+                '<div><label for="id_name">Name</label>'
+                '<input type="text" name="name" maxlength="10" '
+                'required id="id_name"></div><div>'
+                '<label for="id_colour">Colour</label>'
+                '<input type="text" name="colour" maxlength="10" '
+                'required id="id_colour"></div>',
+            )
+
+        with self.subTest("Constructor args takes priority over class attribute"):
+
+            class TestForm(Form):
+                bound_field_class = BoundFieldWithoutColon
+
+                name = CharField(max_length=10)
+                colour = CharField(max_length=10)
+
+            f = TestForm(bound_field_class=BoundFieldWithTwoColons)
+            self.assertHTMLEqual(
+                f.as_div(),
+                '<div><label for="id_name">Name::</label>'
+                '<input type="text" name="name" maxlength="10" '
+                'required id="id_name"></div><div>'
+                '<label for="id_colour">Colour::</label>'
+                '<input type="text" name="colour" maxlength="10" '
+                'required id="id_colour"></div>',
+            )
+
+    def test_field_custom_bound_field(self):
+        class CustomField(CharField):
+            bound_field_class = BoundFieldWithoutColon
+
+        class TestForm(Form):
+            name = CustomField(max_length=10)
+            colour = CharField(max_length=10)
+            comment = CustomField(bound_field_class=BoundFieldWithTwoColons)
+
+        f = TestForm()
+        self.assertHTMLEqual(
+            f.as_div(),
+            '<div><label for="id_name">Name</label>'
+            '<input type="text" name="name" maxlength="10" '
+            'required id="id_name"></div><div>'
+            '<label for="id_colour">Colour:</label>'
+            '<input type="text" name="colour" maxlength="10" '
+            'required id="id_colour"></div>'
+            '<div><label for="id_comment">Comment::</label>'
+            '<input id="id_comment" name="comment" required type="text"></div>',
+        )
+
 
 @jinja2_tests
 class Jinja2FormsTestCase(FormsTestCase):
     pass
 
 
+class BoundFieldWithoutColon(BoundField):
+    def label_tag(self, contents=None, attrs=None, label_suffix=None, tag=None):
+        return super().label_tag(
+            contents=contents, attrs=attrs, label_suffix="", tag=None
+        )
+
+
+class BoundFieldWithTwoColons(BoundField):
+    def label_tag(self, contents=None, attrs=None, label_suffix=None, tag=None):
+        return super().label_tag(
+            contents=contents, attrs=attrs, label_suffix="::", tag=None
+        )
+
+
 class CustomRenderer(DjangoTemplates):
     form_template_name = "forms_tests/form_snippet.html"
     field_template_name = "forms_tests/custom_field.html"
 
 
+class CustomRenderer2(DjangoTemplates):
```

Comment:
```suggestion
class CustomRendererBoundFieldWithoutColon(DjangoTemplates):
```
(want to give this a more descriptive name)